src/Algebra/Graph.hs

```diff
@@ -1000,8 +986,9 @@ simple op x y
 box :: Graph a -> Graph b -> Graph (a, b)
 box x y = overlays $ xs ++ ys
   where
-    xs = map (\b -> fmap (,b) x) $ toList y
-    ys = map (\a -> fmap (a,) y) $ toList x
+    xs = map (\b -> fmap (,b) x) $ toListGr y
+    ys = map (\a -> fmap (a,) y) $ toListGr x
+    toListGr = foldg [] pure (++) (++)
```

Comment:
I suggest to use the name `toList` since the function is equivalent to the standard `toList`.

By the way, are you sure this is fast enough? Do we need to worry about potential quadratic list concatenation?

You might want to use `Algebra.Graph.Internal.List` instead to achieve constant-time list concatenation.

Reply:
Indeed, it is better !